app/components/publications/batch-documents-publication-modal.js

```diff
@@ -94,4 +109,36 @@ export default class PublicationsBatchDocumentsPublicationModalComponent extends
     piece.publicationFlow = null;
     await piece.save();
   }
+
+  async setRegulationTypeThroughReferenceDocument(referenceDocument) {
+    const publicationFlow = await referenceDocument.publicationFlow;
+    let regulationType = await publicationFlow.regulationType;
+    if (!regulationType) {
+      const documentContainer = await referenceDocument.documentContainer;
+      const documentType = await documentContainer.type;
+
+      switch (documentType.uri) {
+        case CONSTANTS.DOCUMENT_TYPES.DECREET:
+          regulationType = await this.store.findRecordByUri(
+            'regulation-type',
+            CONSTANTS.REGULATION_TYPES.DECREET
+          );
+          break;
+        case CONSTANTS.DOCUMENT_TYPES.BVR:
+          regulationType = await this.store.findRecordByUri(
+            'regulation-type',
+            CONSTANTS.REGULATION_TYPES.BVR
+          );
+          break;
+        case CONSTANTS.DOCUMENT_TYPES.MB:
+          regulationType = await this.store.findRecordByUri(
+            'regulation-type',
+            CONSTANTS.REGULATION_TYPES.MB
+          );
+          break;
+      }
+      publicationFlow.regulationType = regulationType;
+      await publicationFlow.save();
+    }
+  }
```

Comment:
Het mappen van document-type >< regulation-type zou apart kunnen staan van het effectief opslaan van publication-flow.
Je geeft document-type mee als params, en indien die mapt komt er een regulation-type uit.
Die kan je dan meegeven aan de creatie van publicationFlow.

Bij edit check je dan eerst het bestaan van reg-type, method aanroepen indien nodig, en dan een set/save in linkPublicationFlow indien er een reg-type terug kwam.